proposals/0000-symbol-newtype-string.rst

```diff
@@ -0,0 +1,134 @@
+Make Symbol a newtype over String
+=================================
+
+.. author:: Oleg Grenrus (with amendments by Chris Dornan and Adam Gundry)
+.. date-accepted::
+.. ticket-url::
+.. implemented::
+.. highlight:: haskell
+.. header:: This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/562>`_.
+.. sectnum::
+.. contents::
+
+I propose changing ``data Symbol`` definition
+to be
+
+::
+
+    newtype Symbol = MkSymbol String
+
+Motivation
+----------
+
+The motivation is similar to accepted and implemented
+`Unify Nat and Natural proposal <https://github.com/ghc-proposals/ghc-proposals/pull/364>`_.
+
+Consider the data type with fields of type ``String``::
+
+  data Fields = Fields [String]
+
+``Fields`` is inhabited by terms, and since recently (`The Char Kind <https://github.com/ghc-proposals/ghc-proposals/pull/387>`_) also by 
+types. However type literals produce ``Symbol``, not ``String`` ::
+
+  fs = Fields ["foo", "bar"]           -- ok
+  type FS = Fields '[ "foo", "bar" ]   -- not ok
+
+Alternatively, we could declare the type with fields of type ``Symbol``::
+
+  data Fields = Fields [Symbol]
+
+Then it would have the opposite issue::
+
+  fs = Fields ["foo", "bar" ]         -- not ok
+  type FS = Fields '[ "foo", "bar" ]  -- ok
+
+As with ``Nat`` and ``Natural`` proposal, it's possible to go roundabout
+this (and e.g. ``singletons-base`` does). However, if ``Symbol`` were
+an ordinary ``newtype`` over ``String``, we would avoid this issue entirely.
+
+(Note: ``singletons-base`` demotes ``Symbol`` to ``Text`` which is
+incorrect, as ``Text`` cannot represent all ``String`` values)
+
+(Note: An API for working with Symbol at the term level is out of scope for this proposal,
+but may be the subject of a subsequent CLC proposal.)
+
+Proposed Change Specification
+-----------------------------
+
+* Change ``data Symbol`` to ``newtype Symbol = MkSymbol String``
+
+The definition being changed lives in ``GHC.Types``, and thus accessing the newtype constructor
+requires importing ``ghc-prim``.
+
+We don't propose any changes to ``GHC.TypeLits`` interface.
+Firstly the implementation of ``Symbol`` stays internal.
+Also ``symbolVal`` will continue to return ``String`` value.
+In future the ``GHC.Symbol`` interface mirroring ``GHC.TypeLits`` functionlity
+could be vreated , but using ``Symbol`` also for the term level can be made as
```

Comment:
s/vreated/revisited? I'm not 100% what word it is supposed to be

Reply:
'created'